ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -236,8 +238,13 @@ protected Node visitJoin(final Join node, final AnalysisContext context) {
     JoinOn joinOn = (JoinOn) (node.getCriteria().get());
```

Comment:
This line here will throw an exception if the criteria is not present

Reply:
Added a check for this.